core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -678,7 +683,13 @@ Builder populate2() {
       final RexCallImplementor value = new ValueConstructorImplementor();
       map.put(MAP_VALUE_CONSTRUCTOR, value);
       map.put(ARRAY_VALUE_CONSTRUCTOR, value);
+
+      // Each subscript operator has a corresponding offset and 'safe' value.
       map.put(ITEM, new ItemImplementor());
+      map.put(OFFSET, new ItemImplementor());
```

Comment:
It would be better to use a single instance of ItemImplementor instead of creating separate instances for each operator. Apart from that, everything else looks good.